docs/code/IDataViewDesignPrinciples.md

```diff
@@ -0,0 +1,479 @@
+  components. The AzureML Algorithms team has implemented loaders and savers
+  for some binary and text file formats, but additional loaders and savers can
+  (and will) be implemented. In particular, implementing a loader from XDF
+  will be straightforward. Implementing a saver to XDF will likely require the
+  XDF format to be extended to support vector-valued columns.
+
+* **Multi-node computation over multiple data partitions**: The IDataView
+  design is focused on single node computation. We expect that in multi-node
+  applications, each node will be given its own data partition(s) to operate
+  on, with aggregation happening outside an IDataView pipeline.
+
+## Schema and Type System
+
+### Basics
+
+IDataView has general schema support, in that a view can have an arbitrary
+number of columns, each having an associated name, index, data type, and
+optional metadata.
+
+Column names are case sensitive. Multiple columns can share the same name, in
+which case, one of the columns hides the others, in the sense that the name
+will map to one of the column indices, the visible one. All user interaction
+with columns should be via name, not index, so the hidden columns are
+generally invisible to the user. However, hidden columns are often useful for
+diagnostic purposes.
+
+The set of supported column data types forms an open type system, in the sense
+that additional types can be added at any time and in any assembly. However,
+there is a precisely defined set of standard types including:
+
+* Text
+* Boolean
+* Single and Double precision floating point
+* Signed integer values using 1, 2, 4, or 8 bytes
+* Unsigned integer values using 1, 2, 4, or 8 bytes
+* Unsigned 16 byte values for ids and probabilistically unique hashes
+* Date time, date time zone, and timespan
+* Key types
+* Vector types
+
+The set of standard types will likely be expanded over time.
+
+The IDataView type system is specified in a separate document, *IDataView Type
+System Specification*.
+
+IDataView provides a general mechanism for associating semantic metadata with
+columns, such as designating sets of score columns, names associated with the
+individual slots of a vector-valued column, values associated with a key type
+column, whether a column's data is normalized, etc.
+
+While IDataView schema supports an arbitrary number of columns, it, like most
+schematized data systems, is designed for a modest number of columns,
+typically, limited to a few hundred. When a large number of *features* are
+required, the features should be gathered into one or more vector-valued
+columns, as discussed in the next section. This is important for both user
+experience and performance.
+
+### Vector Types
+
+Machine learning and advanced analytics applications often involve high-
+dimensional data. For example, a common technique for learning from text,
+known as bag-of-words, represents each word in the text as a numeric feature
```

Comment:
Would it be helpful to our users to link Wikipedia-terms like [bag-of-words](https://en.wikipedia.org/wiki/Bag-of-words_model), or more of a distraction?

Reply:
Yeah actually, especially since the point of this paragraph seems to be to explain the concept. Wouldn't hurt to have a backup description at hand in case this brief description is insufficient.

---
In reply to: [188899313](https://github.com/dotnet/machinelearning/pull/173#discussion_r188899313) [](ancestors = 188899313)